core/src/test/java/io/parsingdata/metal/ShorthandsTest.java

```diff
@@ -214,4 +218,28 @@ public void checkSeqTokens() {
         assertEquals(DEFB, seq.tokens.tail.head);
     }
 
+    final ParseGraph PARSEGRAPH = ParseGraph.EMPTY.add(createParseValue("a", 126)).add(createParseValue("a", 84)).add(createParseValue("a", 42));
+
+    @Test
+    public void mapLeftWithSub() {
```

Comment:
Nice! Question: What if the operand to expand has more than one value?

Say I would add two ParseValues both name ```b``` with value 1 and 2 to the graph. If I would then call:
```java
mapLeft(Shorthand::sub, ref("a"), ref("b"))
```
The expand would then expand the right operand to a list of size 6, with alternating values of 1 and 2. The result of sub would then be a list of size 6, with three empties at the end. Would performing the operation on the cartesian product instead make any sense/be useful?